### DISCUSSION TITLE
Google Oauth: Cannot create account using personal gmail address

### DISCUSSION THREAD

**Discussion by avaika** [outline/outline#5960] [Self Hosting]
I just started a brand new outline instance with docker-compose. I'm trying to configure google oauth, but it doesn't work.

I get `"BadRequestError: Cannot create account using personal gmail address` exception every time I'm trying to create an account.

Looking at the code -- https://github.com/outline/outline/blob/9924fa6621df34099e1957fe93dbcf72104ccd35/plugins/google/server/auth/google.ts#L64-L80 -- it looks like it is intentionally requires google workspaces account.

Why? Why can't I create an account using my personal account?

Just in case:

- I saw this discussion https://github.com/outline/outline/discussions/3875 -- and it's not relevant. I have correct 32 bit values.
- oauth is correctly configured on google side and all the secrets are put to docker.env
- Here's how the error looks like in the log:
```
{"label":"http","level":"info","message":"  <-- GET /auth/google.callback?state=<STATE>&code=4%2F<CODE>&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+openid&authuser=1&prompt=consent"}
{"error":"Cannot create account using personal gmail address","level":"error","message":"Error during authentication","stack":"BadRequestError: Cannot create account using personal gmail address\n    at GmailAccountCreationError (/opt/outline/build/server/errors.js:146:34)\n    at Strategy._verify (/opt/outline/build/plugins/google/server/auth/google.js:50:55)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
{"label":"http","level":"info","message":"  --> GET /auth/google.callback?state=<STATE>&code=4%2F<CODE>&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+openid&authuser=1&prompt=consent 302 258ms 135b"}
```

**Comment by avaika** [comment]
For whoever is concerned on how to get it up & running for personal users I found this: https://github.com/outline/outline/issues/1881#issuecomment-1362926117

**Reply by Zerka30** [reply]
Did you sucessfully run it with personnal gmail account ?

Personally I have the following error : 

```
{"label":"http","level":"info","message":"  --> GET /auth/google 302 3ms 0b"}
{"label":"http","level":"info","message":"  <-- GET /auth/google.callback?state=848afcf41d926593&code=<CODE>&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+openid&authuser=0&prompt=consent"}
{"error":"getaddrinfo EAI_AGAIN www.googleapis.com","level":"error","message":"Error during authentication","stack":"Error: getaddrinfo EAI_AGAIN www.googleapis.com\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:108:26)"}
{"label":"http","level":"info","message":"  --> GET /auth/google.callback?state=848afcf41d926593&code=<CODE>&scope=email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+openid&authuser=0&prompt=consent 302 5,016ms 69b"}
{"label":"http","level":"info","message":"  <-- GET /?notice=auth-error"}
{"label":"http","level":"info","message":"  --> GET /?notice=auth-error 200 21ms 1.65kb"}
```

**Reply by ALenfant** [reply]
I managed to make it work!
It turns out the code relies on a team domain being set:

When logging in from Google, the code tries to fetch the team here: https://github.com/outline/outline/blob/9bc1788bc0aa8d82d458c375bcb3fc4027eed834/plugins/google/server/auth/google.ts#L68
Which tries to find the team by domain: https://github.com/outline/outline/blob/9bc1788bc0aa8d82d458c375bcb3fc4027eed834/server/utils/passport.ts#L123

However, the first created team doesn't have a domain!

So you need to run this in your PostgreSQL database:
```sql
UPDATE teams SET domain='[YOUR_DOMAIN_HERE]';
```

With the same domain as your URL: eg. if your `URL=https://mywiki.thedomain.net/`, put `mywiki.thedomain.net`!
